File: lab_4/Ellipse.cpp
#include "Ellipse.hpp"

Ellipse::Ellipse(double a, double b) : Circle(a){
    this->b = b;
    std::cout << "Ellipse constructor with parametrs" << std::endl;
}

Ellipse::Ellipse() {
    this->a = 1;
    this->b = 2;
    std::cout << "Ellipse constructor default" << std::endl;
}

Ellipse::~Ellipse() {
    std::cout << "Ellipse destructor" << std::endl;
}

double Ellipse::countLength() {
    return M_PI * (this->a + this->b);
}

double Ellipse::countSquare() {
    return M_PI * this->a * this->b;
}

void Ellipse::view() {
    std::cout << "Ellipse minor semiaxis = " << this->a << std::endl;
    std::cout << "Ellipse major semiaxis = " << this->b << std::endl;
}

File: lab4/src/Ellipse/Ellipse.java
package Ellipse;

import Circle.Circle;

public class Ellipse extends Circle {
    private double b;

    public Ellipse(double a, double b) {
        super(a);
        this.b = b;
        System.out.println("Ellipse constructor with parametrs");
    }

    public Ellipse() {
        this.a = 1;
        this.b = 2;
        System.out.println("Ellipse constructor default");
    }

    @Override
    public void view() {
        System.out.println("Ellipse minor semiaxis = " + this.a);
        System.out.println("Ellipse major semiaxis = " + this.b);
    }
}
